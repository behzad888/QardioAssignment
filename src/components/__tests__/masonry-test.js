//@flow
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Masonry from '../masonry/masonry';

describe('while masonry test running in a browser environment', () => {
  let container, wrapper;

  beforeEach(() => {
    jest.resetModules();

    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    ReactDOM.render(null, container);
    document.body.removeChild(container);
    container = null;
  });

  it('should render correctly', () => {
    wrapper = renderer
      .create(
        <div>
          <Masonry>
            <div className="child-item-1"></div>
            <div className="child-item-2"></div>
            <div className="child-item-3"></div>
            <div className="child-item-4"></div>
          </Masonry>
        </div>,
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
