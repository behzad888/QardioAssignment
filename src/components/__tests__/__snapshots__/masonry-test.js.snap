// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`while masonry test running in a browser environment should render correctly 1`] = `
<div>
  <div>
    <div
      className="masonry-layout"
    >
      <div
        className="child-item-1"
      />
      <div
        className="child-item-2"
      />
      <div
        className="child-item-3"
      />
      <div
        className="child-item-4"
      />
    </div>
  </div>
</div>
`;
